### PR TITLE
Invoke commands

### PR DESCRIPTION
**Pull request by jjravi**
Renamed LTO specific messages to something more generic -- INVOKE
Added support for a response with INVOKED

### CODE REVIEW

File: client.cc [hunk 1]
```diff
@@ -258,6 +260,17 @@ Packet Client::LTOCompile (char const *const *argv, size_t argc)
   return MaybeRequest (Detail::RC_LTO_COMPILE);
 }
 
+// LTO-RESPONSE $message
+Packet LTOResponse (std::vector<std::string> &words)
+{
+  if (words[0] == u8"LTO-RESPONSE") {
```
Comment (urnathan) [inline client.cc]: don't need the extraneous bracing (and note they're indented incorrectly if we did)

File: cody.hh [hunk 2]
```diff
@@ -577,7 +577,7 @@ public:
 class Server
 {
 public:
-  enum Direction {READING, WRITING, PROCESSING};
+  enum Direction {READING, WRITING, PROCESSING, STALLED};
```
Comment (urnathan) [inline cody.hh]: This isn't right.  Is it for the case when we're waiting on make to do something before we can assemble a response?
If so, that is what the PROCESSING state is for.  I only expected the server to need 3 states.

**Review by urnathan — APPROVED** [review]

**Review by urnathan — APPROVED** [review]

**Review by urnathan — CHANGES_REQUESTED** [review]
IIUC this new state is a property of make's server class, not libcody's

**Review by urnathan — APPROVED** [review]

**Review by urnathan — APPROVED** [review]

File: client.cc (hunk 1 above)
Comment (jjravi) [reply client.cc]: ah yes, old habits..

File: cody.hh (hunk 2 above)
Comment (jjravi) [reply cody.hh]: Yep, that was the primary reason I added another state. I was also using that extra state to track if ProcessRequests() was resolving an InvokeSubProcessRequest, so I know not to call PrepareToWrite() immediately after ProcessRequests(). 

I thought about adding a state member to my derived Cody::Server class. I originally wanted to pass in a derived Cody::Server class instance to InvokeSubProcessRequest, but the compiler kept complaining that the virtual function override expects a Cody::Server class parameter. Any advice?

Comment (urnathan) [reply cody.hh]: you need that information in the client_state object, which IIRC either derives from or has Cody::Server as a member.

**Pull request merged**